src/include/daos_srv/dtx_srv.h

```diff
@@ -84,14 +81,42 @@ struct dtx_handle {
 	struct dtx_id			*dth_dti_cos;
 	/* The identifier of the DTX that conflict with current one. */
 	struct dtx_conflict_entry	*dth_conflict;
-	/* The data attached to the dth for dispatch */
-	struct dtx_exec_arg		*dth_exec_arg;
 	/** The address of the DTX entry in SCM. */
 	umem_off_t			 dth_ent;
 	/** The address (offset) of the (new) object to be modified. */
 	umem_off_t			 dth_obj;
 };
 
+/* Each sub transaction handle to manage each sub thandle */
```

Comment:
sub_handle is on the leader to track each shard commit status. I had thought to use dtx_sub_handle directly, but this will cause confuse with dtx_handle, since this sub_handle only exist on leader node, so I call it dtx_leader_sub_handle, I want to change dtx_handle to dtx_local_handle, but this will cause so much "rename" and conflict, so maybe postpone it when separate undo log and distribute transaction totally.

Reply:
IMO, "handle" is overloaded at here, this is only a data structure to track status of cohorts, we don't need to close/free a "sub-handle", so it should be renamed to "dtx_cohort_status" or something like that

Reply:
Hmm, it is used to track each sub-parties of the distribute transaction, and only exists on leader. Maybe we can call it dtx_sub_status, if you hate handle.

Reply:
The "handle" is some confused, especially we will introduce more 'handle' for the undo logs. So if possible, renaming this one is better.